### DISCUSSION TITLE
조건에 따른 useListener 질문

### DISCUSSION THREAD

**Discussion by zerone-jang** [zeakd/react-naver-maps#108] [Q&A] [answered]
안녕하세요!

useListener를 특정 조건에 따라서 이벤트를 발생시키거나 제거하고 싶습니다.

이벤트 객체를 제거하는 방식으로 접근했는데
useListener의 return이 void라서 이 방식으로는 안될 것 같습니다.

![스크린샷 2023-07-27 오후 4 54 17](https://github.com/zeakd/react-naver-maps/assets/90392240/16acd33b-155d-4c1d-8e10-a7864a8c1ad7)

어떻게 구현할 수 있을까요?

**Comment by zerone-jang** [comment, marked as answer]
아 이런식으로 하면 되는군요..ㅎㅎ

![event](https://github.com/zeakd/react-naver-maps/assets/90392240/85a3d4bc-3aa5-4eef-93b3-3cc43c8ec7ba)

**Reply by zeakd** [reply]
네 맞습니다. 그리고 완전히 event listener를 추가/제거 하도록 컨트롤 하고싶다면
https://zeakd.github.io/react-naver-maps/guides/customize-overlays/
에서 예를 든 것 처럼 Listener 컴포넌트를 사용해도 좋습니다
